plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/KafkaFilterManager.java

```diff
@@ -129,11 +129,15 @@ public KafkaFilteringResult getKafkaFilterResult(
             if (offsetTimestampRanged.isPresent()) {
                 try (KafkaConsumer<byte[], byte[]> kafkaConsumer = consumerFactory.create(session)) {
                     Optional<Range> finalOffsetTimestampRanged = offsetTimestampRanged;
-                    partitionBeginOffsets = overridePartitionBeginOffsets(partitionBeginOffsets,
-                            partition -> findOffsetsForTimestampGreaterOrEqual(kafkaConsumer, partition, finalOffsetTimestampRanged.get().getBegin()));
+                    if (offsetTimestampRanged.get().getBegin() >= 0) {
```

Comment:
It would better to extract a constant for 0. e.g. `MIN_ALLOWED_OFFSET`

Reply:
OK, I used `INVALID_KAFKA_RANGE_INDEX`